client/components/domains/use-your-domain-step/index.jsx

```diff
@@ -172,22 +172,52 @@ class UseYourDomainStep extends React.Component {
 		const productSlug = getDomainProductSlug( searchQuery );
 		const domainsWithPlansOnlyButNoPlan =
 			domainsWithPlansOnly && ( ( selectedSite && ! isPlan( selectedSite.plan ) ) || isSignupStep );
+		const domainProductSalePrice = getDomainTransferSalePrice(
+			productSlug,
+			productsList,
+			currencyCode
+		);
 
-		let domainProductPrice = getDomainPrice( productSlug, productsList, currencyCode );
-		if ( domainProductPrice ) {
-			domainProductPrice += ' per year';
+		if (
+			isEmpty( domainProductSalePrice ) ||
+			isNextDomainFree( cart ) ||
+			isDomainBundledWithPlan( cart, searchQuery ) ||
+			domainsWithPlansOnlyButNoPlan
+		) {
+			return;
 		}
 
+		return translate( 'Sale price is %(cost)s', { args: { cost: domainProductSalePrice } } );
+	};
+
+	getTransferPriceText = () => {
+		const {
+			cart,
+			currencyCode,
+			translate,
+			domainsWithPlansOnly,
+			isSignupStep,
+			productsList,
+			selectedSite,
+		} = this.props;
+		const { searchQuery } = this.state;
+		const productSlug = getDomainProductSlug( searchQuery );
+		const domainsWithPlansOnlyButNoPlan =
+			domainsWithPlansOnly && ( ( selectedSite && ! isPlan( selectedSite.plan ) ) || isSignupStep );
+
+		const domainProductPrice = getDomainPrice( productSlug, productsList, currencyCode );
+
 		if (
 			domainProductPrice &&
 			( isNextDomainFree( cart ) ||
 				isDomainBundledWithPlan( cart, searchQuery ) ||
-				domainsWithPlansOnlyButNoPlan )
+				domainsWithPlansOnlyButNoPlan ||
+				getDomainTransferSalePrice( productSlug, productsList, currencyCode ) )
 		) {
-			domainProductPrice = translate( 'Renews at ' ) + domainProductPrice;
+			return translate( 'Renews at %(cost)s', { args: { cost: domainProductPrice } } );
 		}
 
-		return domainProductPrice;
+		return translate( '%(cost)s per year', { args: { cost: domainProductPrice } } );
```

Comment:
There's a bug here, because sometimes we don't have the `domainProductPrice` set. For example, when no query.

<img width="363" alt="Screenshot 2019-03-15 at 17 34 54" src="https://user-images.githubusercontent.com/1103398/54447064-b916b000-4748-11e9-820b-4f17297e639f.png">

Reply:
Oh yeah.  Good catch.